CMakeLists.txt

```diff
@@ -320,6 +327,15 @@ if (USE_OPENMP)
     endif (OpenMP_CXX_FOUND)
 endif (USE_OPENMP)
 
+# what library is used for bluetooth is currently decided by a preprocessor define at the top of src/utils/bluetooth.cpp
+target_link_libraries (${BOARD_CONTROLLER_NAME} PRIVATE nativeble)
+
+find_package(PkgConfig REQUIRED)
+pkg_search_module(GATTLIB REQUIRED gattlib)
```

Comment:
am I right that it impicates linux and bluez?

This way we will have to implement it for each operation system since ble stack is very different between OSes.

I thought that we should use BLED112 dongle and use BGLIB(serial port in fact), ganglion implementation can be used as a reference.

Also for bluez related api I have a concern about licenses, most of them distributed under GPL which is not compatible with MIT

Reply:
these are just libraries I linked in to get bluetooth working.  nativeble is MIT and cross-platform, but not very stable.  gattlib is GPL and linux-only but is just for testing.  I mentioned these on the muse issue thread near the top.

I didn't realise the BLED112 dongle also provides a full bluetooth interface.  I mistakenly thought it was domain-specific like the cyton dongle.

The interface I wrote to these libraries should be mergeable with the ganglion interface, but I would need to purchase a BLED112 dongle to test.

A notable advantage of general-purpose bluetooth is that users would not have to purchase a specific dongle, but could optionally use the existing bluetooth support of their devices, if they install a library that connects with them.

Reply:
"A notable advantage of general-purpose bluetooth is that users would not have to purchase a specific dongle, but could optionally use the existing bluetooth support of their devices, if they install a library that connects with them."

thats true, when I was working on ganglion my first try was to use OS specific libs, but later I realised that all of them are completely different and all of them have different issues(example windows native SDK in C++ has some bugs, so it doesnt work with C# gui apps and requires CLR to be compiled)

Reply:
given i don't have a bled112 at this time, but am happy to get a bluetooth backend working, any thoughts?  if i migrate the bglib code to a generic interface in src/utils that still uses bglib, would you be interested in testing it with a ganglion?  the problem is that i make a lot of bugs so the test cycle can be very slow.

that would leave some space for adding platform interfaces for people like me without dongles, once the work is put in to make them work.  also opens avenues for mobile devices where dongles are harder but ble is present.

i've also been thinking about labstreaminglayer some.  it seems some work is duplicated between here and labstreaminglayer.